### PR TITLE
DM-11727: Update miniconda baseline to python3.6

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: scripts/newinstall.sh [hunk 1]
```diff
@@ -633,7 +633,7 @@ n8l::python_check() {
 				if [[ $pyverok != true ]]; then
 					cat <<-EOF
 
-					Thanks. After you install Python 2.7 or 3.5 and the required modules,
```
Comment (jhoblitt) [inline scripts/newinstall.sh]: Could you use variable interpolation similar to the above `2 (>=2.${minver2}) or 3 (>=3.${minver3})`? That would eliminate a set of hard coded version strings.

File: scripts/newinstall.sh [hunk 2]
```diff
@@ -567,7 +567,7 @@ else:
 
 #
 #	Test/warn about Python versions, offer to get miniconda if not supported.
-#	LSST currently mandates Python 3.5 and, optionally, 2.7.  We assume that the
+#	LSST currently mandates Python 3.6 and, optionally, 2.7.  We assume that the
```
Comment (jhoblitt) [inline scripts/newinstall.sh]: Perhaps remove the sentence with the hard coded version strings?

File: scripts/newinstall.sh (hunk 1 above)
Comment (timj) [reply scripts/newinstall.sh]: Removed reference to versions completely. This message comes just after we've already reported the version required.

Comment (jhoblitt) [reply scripts/newinstall.sh]: :+1:

**Comment by timj** [comment]
@jhoblitt those rspec tests are *really* annoying. This patch is fine but fails because the test is checking the exact wording of the echo statement. Why are we testing that?

**Comment by timj** [comment]
@jhoblitt can you please fix this test failure?

**Comment by jhoblitt** [comment]
@timj The test that is failing is checking the error message reporting the minimum python versions is in sync with the version parameters to `n8l::pyverok`.  This stdout check can probably be removed now since the script shouldn't be able to lie about the minimum versions after this PR.

**Comment by timj** [comment]
I know why it's failing. I said so above. Please can you fix it as I think we do need to get this update merged and I am too busy today to deal with it the way you probably want it done.

**Comment by jhoblitt** [comment]
I'm not sure why travis is failing.  I just tired this branch locally and it passes.

**Comment by timj** [comment]
One last try. I haven't got this stuff running locally so I'm iterating with travis.

**Pull request merged**